### ISSUE TITLE
test87 fails under python 3.6

### ISSUE THREAD

**Issue by Tosainu** [vim/vim#1359]
When I was building Vim with `--with-python3`, `test87` failed with following messages.

```
rm -rf test87.failed test.ok test.out X* viminfo
cp test87.ok test.ok
# Sleep a moment to avoid that the xterm title is messed up.
# 200 msec is sufficient, but only modern sleep supports a fraction of
# a second, fall back to a second if it fails.
VIMRUNTIME=../../runtime; export VIMRUNTIME;  ../vim -f  -u unix.vim -U NONE --noplugin --not-a-term -s dotest.in test87.in
# For flaky tests retry one time.  No tests at the moment.
#@/bin/sh -c "if test -f test.out -a test87 = test61; then \
#	  if diff test.out test87.ok; \
#	  then echo flaky test ok first time; \
#	  else rm -rf test87.failed test.out X* viminfo; \
#		VIMRUNTIME=../../runtime; export VIMRUNTIME;  ../vim -f  -u unix.vim -U NONE --noplugin --not-a-term -s dotest.in test87.in; \
#	  fi \
#	fi"
# Check if the test.out file matches test.ok.
661c661
< sys.stdout.write(None):(<class 'TypeError'>, TypeError('argument must be str, bytes or bytearray, not None',))
---
> sys.stdout.write(None):(<class 'TypeError'>, TypeError("Can't convert 'NoneType' object to str implicitly",))
664c664
< sys.stdout.writelines([1]):(<class 'TypeError'>, TypeError('argument must be str, bytes or bytearray, not int',))
---
> sys.stdout.writelines([1]):(<class 'TypeError'>, TypeError("Can't convert 'int' object to str implicitly",))
703,704c703,704
< import xxx_no_such_module_xxx:(<class 'ModuleNotFoundError'>, ModuleNotFoundError("No module named 'xxx_no_such_module_xxx'",))
< import failing_import:(<class 'ModuleNotFoundError'>, ModuleNotFoundError("No module named 'failing_import'",))
---
> import xxx_no_such_module_xxx:(<class 'ImportError'>, ImportError('No module named xxx_no_such_module_xxx',))
> import failing_import:(<class 'ImportError'>, ImportError('No module named failing_import',))
rm -rf X* test.ok viminfo
```

- Vim v8.0.0147
- Arch Linux
- gcc (GCC) 6.2.1 20160830
- Python 3.6.0

**Comment by anatol** [comment]
@Tosainu thanks for reporting it. I also see the same test failure with python 3.6

**Comment by Jesin** [comment]
[Here is a patch](https://git.archlinux.org/svntogit/packages.git/plain/trunk/fix-tests-python36.patch?h=packages/vim&id=3baf2369615e12babf54001aad2cd8c2b57972ed) that Arch Linux uses to fix this particular issue, though applying it verbatim may cause this test to fail for Python 3 versions prior to 3.6.

**Comment by brammool** [comment]
I'm sending patch 8.0.0422 that should fix this.
However, I didn't manage to run Python 3.6, thus could not verify if the fix works.
If not, please re-open the issue and tell me what still fails.
Or better: fix it :-)

**Comment by jonathonf** [comment]
Confirming this fixes the build issue on Arch with Python 3.6.

(Arch users should comment out/remove the 'fix-tests-python36.patch' patch line from the PKGBUILD otherwise it will now cause a test 87 failure)
